src/main.rs

```diff
@@ -122,16 +129,25 @@ fn run() -> Result<()> {
     // Remember the time we started.
     let start_time = now();
 
-    // Build a regex containing our `--null` value.
-    let null_re = if let Some(null_re_str) = opt.null.as_ref() {
-        // Always match the full CSV value.
-        let s = format!("^{}$", null_re_str);
-        let re = Regex::new(&s).context("can't compile regular expression")?;
-        Some(re)
+    let mut pattern = if opt.null_byte_as_empty == true {
```

Comment:
I'm sure there's a better way to deal with this, but I'm not familiar enough to tell for sure.